### PR TITLE
Use P1 inverted color to mark selected bar in editor

### PR DESCRIPTION
**Pull request by DeinAlptraum**
Before, the selected bar would use P2's color to mark the selected bar. This leads to no marking at all when no color has been selected for P2.

The color attribute for each bar currently holds only a single integer for the player number, since this reuses the coloring-mechanism of the actual sing mode. I chose to mark the selected syllable as belonging to player 99 (instead of player 2) and then making a distinction for that number. The choice is arbitrary, I just didn't want to use 0 since it seems to be already used for something else.

Currently, the color used for marking is the inversion of P1's color. We could also change this to not be based on the player colors at all, and instead use two fixed colors for selected/not selecte or similar.

### CODE REVIEW

**Review by barbeque-squared — CHANGES_REQUESTED** [review]
I'm fine with hardcoding/(ab)using 99 to be this color (otherwise it's probably going to explode in PR size) however I would request the following changes:

* define a constant eg `INVERTED_1 = 99` in UDraw (use a better name if you have one)
* Use that constant instead of hardcoding 99 in UDraw and UScreenEditSub

makes things more self-explanatory when someone looks at this code in a year's time or so.

**Comment by DeinAlptraum** [comment]
That's a good idea, changed

File: src/screens/views/UScreenSingView.pas [hunk 1]
```diff
@@ -620,6 +620,15 @@ constructor TScreenSingView.Create;
     Tex_ScoreBG[I - 1] := Texture.LoadTexture(Skin.GetTextureFileName('ScoreBG'), TEXTURE_TYPE_COLORIZED, Color);
   end;
 
+  // Inversions of first player colors used to mark selected note in editor
+  Color := RGBFloatToInt(1 - Col[1].R, 1 - Col[1].G, 1 - Col[1].B);
+  Tex_Left_Inv := Texture.LoadTexture(Skin.GetTextureFileName('GrayLeft'), TEXTURE_TYPE_COLORIZED, Color);
+  Tex_Mid_Inv := Texture.LoadTexture(Skin.GetTextureFileName('GrayMid'), TEXTURE_TYPE_COLORIZED, Color);
+  Tex_Right_Inv := Texture.LoadTexture(Skin.GetTextureFileName('GrayRight'), TEXTURE_TYPE_COLORIZED, Color);
+  Tex_Left_Rap_Inv := Texture.LoadTexture(Skin.GetTextureFileName('GrayLeftRap'), TEXTURE_TYPE_COLORIZED, Color);
```
[the remark this answers is not in the document]
Comment (DeinAlptraum) [reply src/screens/views/UScreenSingView.pas]: Fixed the whitespace and the color change for track changing on duet songs.

**Review by barbeque-squared — APPROVED** [review]

**Comment by barbeque-squared** [comment]
Took me a while to realize that `870.patch` is not the same as `870.diff`. Which makes sense I guess.
And the duet stuff is now also consistent, very nice, thanks!

I've opened #873 to keep track of the whitespace.

**Pull request merged**